packages/plugins/src/plugins/morpho/rewardsFetcher.ts

```diff
@@ -0,0 +1,115 @@
+import {
+  EvmNetworkIdType,
+  NetworkId,
+  PortfolioElementMultiple,
+  PortfolioElementType,
+  formatEvmAddress,
+  getUsdValueSum,
+} from '@sonarwatch/portfolio-core';
+
+import BigNumber from 'bignumber.js';
+import { Address, getAddress } from 'viem';
+import { Fetcher, FetcherExecutor } from '../../Fetcher';
+import { platformId } from './constants';
+import { Cache } from '../../Cache';
+import tokenPriceToAssetToken from '../../utils/misc/tokenPriceToAssetToken';
+import { MorphoRewardsRes } from './types';
+import { getRewards } from './utils/api';
+
+/* 
+    As of April 2025, Morpho rewards are calculated off-chain.
+    According to the Morpho team, the only way to access these rewards is 
+    via the Morpho API, which is subject to a rate limit of 850 requests per minute.
+    The exact formula used to compute rewards is currently closed-source, though the 
+    team has expressed intentions to open-source it in the future.
+*/
+
+function calculateRawRewardBalances(data: MorphoRewardsRes['data']) {
+  const balances: Record<Address, bigint> = {};
+
+  for (const entry of data) {
+    const address = entry.asset?.address
+      ? getAddress(formatEvmAddress(entry.asset?.address))
+      : null;
+    if (!address) continue;
+
+    let raw: bigint | null = null;
```

Comment:
We should use `undefined` instead of `null`.

```suggestion
    let raw: bigint | undefined;
```